src/voter.rs

```diff
@@ -441,16 +465,203 @@ impl<H, N, E: Environment<H, N>> Future for BackgroundRound<H, N, E> where
 
 	fn poll(&mut self) -> Poll<u64, E::Error> {
 		self.task = Some(::futures::task::current());
-		self.inner.poll()?;
 
-		if self.is_done() {
-			Ok(Async::Ready(self.inner.votes.number()))
+		let mut voting_round = self.inner.lock();
+		voting_round.poll()?;
+
+		if self.is_done(&voting_round) {
+			Ok(Async::Ready(voting_round.votes.number()))
 		} else {
 			Ok(Async::NotReady)
 		}
 	}
 }
 
+pub struct RoundCommitter<H, N, E: Environment<H, N>> where
+	H: Hash + Clone + Eq + Ord + ::std::fmt::Debug,
+	N: Copy + BlockNumberOps + ::std::fmt::Debug,
+{
+	voting_round: Arc<Mutex<VotingRound<H, N, E>>>,
+	commit_timer: E::Timer,
+	last_commit: Option<Commit<H, N, E::Signature, E::Id>>,
+}
+
+impl<H, N, E: Environment<H, N>> RoundCommitter<H, N, E> where
+	H: Hash + Clone + Eq + Ord + ::std::fmt::Debug,
+	N: Copy + BlockNumberOps + ::std::fmt::Debug,
+{
+	fn import_commit(
+		&mut self,
+		env: &E,
+		commit: Commit<H, N, E::Signature, E::Id>,
+	) -> Result<bool, E::Error> {
```

Comment:
i'm not sure if there is a check in this function against equivocations.

Is it possible to reuse the vote-graph and round logic for these checks somehow?